### ISSUE TITLE
Alternate Port/Authentication Support

### ISSUE THREAD

**Issue by FerretBomb** [davidborzek/streamdeck-foobar2000#8]
Currently the plugin lacks the ability to use Beefweb's built-in authentication, or to specify a non-default port or remote IP for multi-system configurations.
Would be appreciated if these could be added. Thanks in advance!

**Comment by MidnightAnnie** [comment]
Yeah would really like to be able to activate the authentication without breaking the stream deck plugin

**Comment by ChristopherW** [comment]
Agreed, I already use Beefweb on a non-standard port with Beefweb's built-in authentication, and the plugin expects hardcoded 8880 with no auth for the API calls. I tried setting the port and auth settings to ones the plugin is hardcoded to expect, using Beefweb 0.5 it still failed to communicate with fb2k (Windows 11, v6.6.1 Stream Deck software). Upgrading to Beefweb 0.8 seems to have fixed that issue, though I presume this project may be abandonware if the dev is too busy to add support for HTTP basic authentication. 

I've come up with alternatives for basic player controls by using global button binds and Windows' inbuilt virtual F13-F24 keys, called via SD dials or button pushes. F13-F24 are normally referenced as Shift+F1-F12; the shift essentially adds 12. However fb2k intercepts the Shift keypress when setting up keybinds. I ended up first assigning virtual F keys to buttons and dials in Stream Deck using Hotkey instances, then using those to 'type' in to new fb2k keyboard binds in foobar2000's Keyboard Shortcuts page. Works fine for what I need.
